zarr/hierarchy.py

```diff
@@ -414,6 +416,78 @@ def arrays(self):
                                  chunk_store=self._chunk_store,
                                  synchronizer=self._synchronizer)
 
+    def visit(self, func):
+        """Run callable on each object name.
+
+        Examples
+        --------
+        >>> from __future__ import print_function, unicode_literals
+        >>> import zarr
+        >>> g1 = zarr.group()
+        >>> g2 = g1.create_group('foo')
+        >>> g3 = g1.create_group('bar')
+        >>> d1 = g1.create_dataset('baz', shape=100, chunks=10, fill_value=0)
+        >>> d2 = g1.create_dataset('quux', shape=200, chunks=20, fill_value=0)
+        >>> def print_visitor(name):
+        ...     print(repr(name))
+        >>> g1.visit(print_visitor)
+        bar
+        baz
+        foo
+        quux
+
+        """
+
+        def _visit(obj, func=func):
+            keys = sorted(getattr(obj, "keys", lambda : [])())
+            for each_key in keys:
+                for value in _visit(obj[each_key], func=func):
```

Comment:
Yeah, I was playing with stuff and pushed a bad commit. Unfortunately have been AFK until now. Should be fixed.